extensions/cache/src/main/java/org/radargun/stages/cache/generators/RandomStreamGenerator.java

```diff
@@ -0,0 +1,141 @@
+package org.radargun.stages.cache.generators;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Random;
+
+import org.radargun.config.DefinitionElement;
+
+/**
+ * Generator for {@link RandomStream} objects
+ * 
+ * @author zhostasa
+ */
+@DefinitionElement(name = "randomStream", doc = "Generates stream of random data, specifically for stream operation testing")
+public class RandomStreamGenerator implements ValueGenerator {
+
+   public Object generateValue(Object key, int size, Random random) {
+      return new RandomStream(key.hashCode(), size);
+   }
+
+   /**
+    * Will attempt to retrieve length of either {@link RandomStream} or byte[]
+    * 
+    * @return Size of object or 0
+    */
+   public int sizeOf(Object value) {
+      if (value instanceof RandomStream)
+         return ((RandomStream) value).length;
+      if (value instanceof byte[])
+         return ((byte[]) value).length;
+      return 0;
+   }
+
+   /**
+    * If value is byte[] then creates new {@link RandomStream} and validates it
+    * against it
+    * 
+    * @param value
+    *           Should be byte[]
+    * @param key
+    *           Hashcode will he used as seed for {@link RandomStream}
+    * @param expectedSize
+    *           Expected size of data
+    * 
+    * @return true if value is byte[] and validation succeeded
+    */
+   public boolean checkValue(Object value, Object key, int expectedSize) {
+      boolean ret;
+      if (value instanceof byte[]) {
+         RandomStream rand = new RandomStream(key.hashCode(), expectedSize);
```

Comment:
1) creating a stream and closing that should be done using try-with-resources construct
2) I suggest moving the contents of `validate` to this method, rather than creating a new instance of `RandomStream` in order to execute the instance-method
3) I am wondering what is the code path that leads to executing this method with `byte[]` argument, since if the RSG created `RandomStream`, it should check `RandomStream`s again (or at least `InputStream`s).